mailpoet/lib/Automation/Integrations/WooCommerce/Fields/OrderFieldsFactory.php

```diff
@@ -0,0 +1,347 @@
+          'woocommerce:order:shipping-state',
+          Field::TYPE_STRING,
+          __('Shipping state/county', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_shipping_state();
+          }
+        ),
+        new Field(
+          'woocommerce:order:shipping-country',
+          Field::TYPE_STRING,
+          __('Shipping country', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_shipping_country();
+          }
+        ),
+        new Field(
+          'woocommerce:order:created-date',
+          Field::TYPE_DATETIME,
+          __('Created date', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_date_created();
+          }
+        ),
+        new Field(
+          'woocommerce:order:paid-date',
+          Field::TYPE_DATETIME,
+          __('Paid date', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_date_paid();
+          }
+        ),
+        new Field(
+          'woocommerce:order:customer-note',
+          Field::TYPE_STRING,
+          __('Customer provided note', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_customer_note();
+          }
+        ),
+        new Field(
+          'woocommerce:order:payment-method',
+          Field::TYPE_ENUM,
+          __('Payment method', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_payment_method();
+          },
+          [
+            'options' => $this->getOrderPaymentOptions(),
+          ]
+        ),
+        new Field(
+          'woocommerce:order:status',
+          Field::TYPE_ENUM,
+          __('Status', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_status();
+          },
+          [
+            'options' => $this->getOrderStatusOptions(),
+          ]
+        ),
+        new Field(
+          'woocommerce:order:total',
+          Field::TYPE_NUMBER,
+          __('Total', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return (float)$payload->getOrder()->get_total();
+          }
+        ),
+        new Field(
+          'woocommerce:order:coupons',
+          Field::TYPE_ENUM_ARRAY,
+          __('Used coupons', 'mailpoet'),
+          function (OrderPayload $payload) {
+            return $payload->getOrder()->get_coupon_codes();
+          },
+          [
+            'options' => $this->getCouponOptions(),
+          ]
+        ),
+        new Field(
+          'woocommerce:order:is-first-order',
+          Field::TYPE_BOOLEAN,
+          __('Is first order', 'mailpoet'),
+          function (OrderPayload $payload) {
+            $order = $payload->getOrder();
+            return !$this->previousOrderExists($order);
+          }
+        ),
```

Comment:
Maybe instead of the query we could utilize `$customer->get_order_count()`

Reply:
@websupporter That's not equivalent, right? Imagine the field is used in a condition after some delays and stuff. What if more orders were created in the meantime? Then the first order is still the first one, I guess.

Reply:
Right! Thanks for clarifying.